src/LabLog/Controllers/AdminController.cs

```diff
@@ -22,7 +22,17 @@ public AdminController(EventModelContext db)
 
         public IActionResult Index()
         {
-            ViewData["Rooms"]="";
+            List<RoomModel> roomList = new List<RoomModel>();
+            var events = from _events in _db.LabEvents
+                         where _events.EventType.Equals(RoomCreatedEvent.EventTypeString)
+                         select _events;
+            foreach (LabEvent roomCreatedEvent in events)
+            {
+                RoomModel room = new RoomModel();
+                room.ApplyRoomCreatedEvent(roomCreatedEvent);
+                roomList.Add(room);
+            }
+            ViewData["Rooms"] = roomList;

```

Comment:
ViewData isn't the best way to communicate with the view. Build a ViewModel class and pass it into `View();`